core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataSplitter.scala

```diff
@@ -59,30 +62,69 @@ case object DataSplitter {
  *
  * @param uid
  */
-class DataSplitter(uid: String = UID[DataSplitter]) extends Splitter(uid = uid) {
+class DataSplitter(uid: String = UID[DataSplitter]) extends Splitter(uid = uid) with DataSplitterParams {
 
   /**
-   * Function to set parameters before passing into the validation step
-   * eg - do data balancing or dropping based on the labels
+   * Function to set the down sampling fraction and parameters before passing into the validation step
    *
    * @param data
    * @return Parameters set in examining data
    */
   override def preValidationPrepare(data: Dataset[Row]): PrevalidationVal = {
-    summary = Option(DataSplitterSummary())
+    val dataSetSize = data.count().toDouble
+    val sampleF = getMaxTrainingSample / dataSetSize
+    val DownSampleFraction = if (getMaxTrainingSample < dataSetSize) sampleF else 1
```

Comment:
could be a `min` here too

also: lowerCamelCase

```suggestion
    val downSampleFraction = math.min(sampleF, 1.0)
```